File: workflow/plugins/go/list/list_concat/list_concat.go
// Package list_concat provides a workflow plugin for concatenating lists.
package list_concat

// ListConcat implements the NodeExecutor interface for concatenating lists.
type ListConcat struct {
	NodeType    string
	Category    string
	Description string
}

// NewListConcat creates a new ListConcat instance.
func NewListConcat() *ListConcat {
	return &ListConcat{
		NodeType:    "list.concat",
		Category:    "list",
		Description: "Concatenate multiple lists",
	}
}

// Execute runs the plugin logic.
func (p *ListConcat) Execute(inputs map[string]interface{}, runtime interface{}) map[string]interface{} {
	lists, ok := inputs["lists"].([]interface{})
	if !ok {
		return map[string]interface{}{"result": []interface{}{}}
	}

	var result []interface{}
	for _, lst := range lists {
		if arr, ok := lst.([]interface{}); ok {
			result = append(result, arr...)
		}
	}

	return map[string]interface{}{"result": result}
}

File: workflow/plugins/python/list/list_concat/list_concat.py
"""Workflow plugin: concatenate lists."""

from ...base import NodeExecutor


class ListConcat(NodeExecutor):
    """Concatenate multiple lists."""

    node_type = "list.concat"
    category = "list"
    description = "Concatenate multiple lists"

    def execute(self, inputs, runtime=None):
        array = inputs.get("array", inputs.get("list", []))
        lists = inputs.get("lists", inputs.get("arrays", [array]))
        result = []
        for lst in lists:
            if isinstance(lst, list):
                result.extend(lst)
        return {"result": result}
